test/e2e/upgradetest/recovery_test.go

```diff
@@ -0,0 +1,74 @@
+// Copyright 2017 The etcd-operator Authors
```

Comment:
It will become a long file by the end with all 5 upgrade test cases.
The issue is that the per test setup/cleanup is a little long and is mostly repeated for each case. Perhaps if we just abstract it we can shorten the test cases and put them in one file. 
The current setup/cleanup being:
```
	defer func() {
		err := testF.DeleteOperator()
		if err != nil {
			t.Fatal(err)
		}
		if err = e2eutil.WaitUntilDeploymentDeleted(testF.KubeCli, 3*time.Second, 30*time.Second, "etcd-operator", testF.KubeNS); err != nil {
			t.Fatalf("failed to delete deployment in time: %v", err)
		}
	}()
	err := testF.CreateOperator()
	if err != nil {
		t.Fatal(err)
	}
	if err = k8sutil.WaitEtcdTPRReady(testF.KubeCli.CoreV1().RESTClient(), 3*time.Second, 30*time.Second, testF.KubeNS); err != nil {
		t.Fatalf("failed to see cluster TPR get created in time: %v", err)
	}

	testClus, err := e2eutil.CreateCluster(t, testF.KubeCli, testF.KubeNS, e2eutil.NewCluster("upgrade-test-", 3))
	if err != nil {
		t.Fatal(err)
	}
	defer func() {
		if err := e2eutil.DeleteCluster(t, testF.KubeCli, testClus, &e2eutil.StorageCheckerOptions{}); err != nil {
			t.Fatal(err)
		}
	}()
	_, err = e2eutil.WaitUntilSizeReached(t, testF.KubeCli, 3, 60*time.Second, testClus)
	if err != nil {
		t.Fatal(err)
	}
	err = testF.UpgradeOperator()
	if err != nil {
		t.Fatal(err)
	}
```
Let me know if you're still sure we should put all the cases in one file?